### ISSUE TITLE
Xcode 8.1: GeneratedMocks build fails "Contextual closure type '(IN) -> OUT' expects 1 argument, but 3 were used in closure body"

### ISSUE THREAD

**Issue by che1404** [SwiftKit/Cuckoo#73]
Hi!
First off, I want to thank you for this awesome library, one of the fewest (and probably the best) we can use effectively with Swift.

When I upgraded to Xcode 8.1 I started having this build error in my GeneratedMocks.swift. 
The mocked protocol is:

```
protocol LoginAPIDataManagerInputProtocol: class
{
    func login(username user: String, password pass: String, completion: @escaping (_ success: Bool, _ error: NSError?) -> Void)
}
```

Also, the same error with: 
```
class LoginAPIDataManager: LoginAPIDataManagerInputProtocol  {

    init() {}
    
    func login(username user: String, password pass: String, completion: @escaping (_ success: Bool, _ error: NSError?) -> Void) {
        completion(true, nil)
    }
}
```

The generated mock code for the first error is in the following func:
```

    func login(username user: String, password pass: String, completion: @escaping (_ success: Bool, _ error: NSError?) -> Void) {
        return manager.call("login(username: String, password: String, completion: @escaping (_ success: Bool, _ error: NSError?) -> Void)", parameters: (user, pass, completion), original: observed.map { o in return { (user: String, pass: String, completion: @escaping (_ success: Bool, _ error: NSError?) -> Void) in o.login(username: user, password: pass, completion: completion) } })
    }

```

My Cuckoo version is 0.7.0.
Any idea on what could be going on?
If I switch to Xcode 8, there are no errors.

Thanks in advance,
Robert.

**Comment by TadeasKriz** [comment]
Hi, I'm glad you like our library and thank you for your report.

What error are you getting? Could you paste it here in full? And is it Cuckoo generator error or compile error in Xcode?

Thanks

**Comment by TadeasKriz** [comment]
I've done some fixes for Xcode 8.1 and released version `0.8.0` to CocoaPods. Please give it a spin. Thanks!

**Comment by che1404** [comment]
Thanks @TadeasKriz ! It worked! Keep up the good work!
